### PR TITLE
Simplify autoShell Action Wiring

### PR DESCRIPTION
**Pull request by TalZaccai**
# Simplify autoShell action wiring

## Summary

This PR redesigns how the TypeAgent desktop agent communicates with autoShell, replacing fragile hand-wired plumbing with a schema-driven, type-safe architecture. Adding a new action drops from ~6 file touches to 1–2.

## Motivation

1. **Massive boilerplate** — `connector.ts` had a 70-case switch where ~60 cases just forwarded JSON to autoShell. On the C# side, 31 of 46 settings actions followed three simple patterns but each was hand-coded.
2. **No error communication** — autoShell wrote results to stdout with no structured response. The TS side had no way to know if an action succeeded, failed, or why.
3. **Silent schema drift** — three independent declaration points (TS schema, connector switch, C# `SupportedCommands`) with no enforcement they agree. 7 C# actions had no TS schema at all.
4. **Unsafe parameter handling** — handlers manually extracted parameters from raw JSON with string-keyed lookups and no type safety, broad `catch (Exception)` blocks, and a hardcoded `bin/Debug` exe path.

### 1. Structured request-response protocol
- `IActionHandler.Handle` now returns `ActionResult` with `Ok()`/`Fail()`/`Quit()` factories and optional `Data` payload — errors propagate back to the TS agent
- `connector.ts` uses GUID-correlated `sendAction` instead of fire-and-forget writes
- The 70-case switch is replaced by a 10-case preprocessor (only actions needing TS-side work)

### 2. Newtonsoft.Json → System.Text.Json
- `Newtonsoft.Json` dependency removed
- Handler interface simplified from `Handle(string key, string value, JToken rawValue)` to `Handle(string key, JsonElement parameters)`

### 3. Schema-driven validation
- `SchemaValidator` parses `.pas.json` schemas and warns at startup about wiring mismatches between TS schemas and C# handlers
- `connector.ts` loads known action names from schemas for runtime validation

### 4. Registration-driven settings handlers
- `SettingsHandlerBase` provides three declarative patterns: `AddRegistryToggleAction`, `AddRegistryMapAction`, `AddOpenSettingsAction`
- 31 settings actions reduced to one-line registrations; 9 handlers refactored (**~785 lines removed**)

### 5. Unified handler hierarchy
- `ActionHandlerBase` with `AddAction(name, handler)` dictionary dispatch replaces static arrays + switch statements across all 17 handlers
- `SettingsHandlerBase` extends `ActionHandlerBase` (single hierarchy)

### 6. Roslyn source generator for typed parameters
- `autoShell.Generators` project reads `.pas.json` schemas at build time and generates C# `record` classes with `[JsonPropertyName]` attributes
- `AddAction<T>(name, handler)` auto-deserializes `JsonElement` to typed records — **33 actions migrated** from manual string lookups to compile-time type safety
- Supports primitives, nullable types, string unions, and arrays

### 7. TypeScript schema additions
- Added missing schemas: `ApplyThemeAction`, `SystemThemeModeAction`, optional `refreshRate` on `SetScreenResolutionAction`
- Regenerated `.pas.json` files so the C# generator picks them up automatically

### 8. Error handling hardening
- Deserialization: try/catch + null guard in `AddAction<T>` returns `ActionResult.Fail` on malformed input
- Registry/process: targeted exception catches (`UnauthorizedAccessException`, `SecurityException`, `IOException`, `Win32Exception`) instead of broad `catch (Exception)`
- Input validation: negative int-to-uint overflow fixed in `SetScreenResolution`
- `connector.ts`: autoShell.exe resolved from Debug/Release with `AUTOSHELL_PATH` env var override

## Impact

| Metric | Before | After |
|---|---|---|
| `connector.ts` switch cases | ~70 | 10 |
| Hand-coded settings actions | 46 | 15 |
| Files to touch for new action | ~6 | 1–2 |
| Error feedback to TS agent | None (stdout only) | Structured `ActionResult` |
| Parameter type safety | Manual string lookups | Generated typed records (33 actions) |
| Test count | 119 | 213 |

## Testing

- **213 tests passing** (0 warnings, 0 failures)
- Covers: handler actions, registered patterns (toggle/map/open-settings), schema validation, dispatcher routing, typed deserialization, edge cases
- E2E tests filtered out (`Category!=E2E`) — require a running desktop

### CODE REVIEW

File: dotnet/autoShell/ActionDispatcher.cs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        }
+
+        if (string.Equals(actionName, "quit", StringComparison.OrdinalIgnoreCase))
+        {
+            return ActionResult.Quit();
+        }
+
+        JsonElement parameters = root.TryGetProperty("parameters", out JsonElement p)
+            ? p
+            : default;
```
[the remark this answers is not in the document]
Comment (TalZaccai) [reply dotnet/autoShell/ActionDispatcher.cs]: Fixed in 5f1e70d5 — now defaults to `JsonDocument.Parse("{}").RootElement.Clone()` instead of `default(JsonElement)`.

File: dotnet/autoShell/Handlers/ActionHandlerBase.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            try
+            {
+                typed = JsonSerializer.Deserialize<T>(parameters.GetRawText(), CamelCaseOptions);
+            }
+            catch (JsonException ex)
+            {
+                return ActionResult.Fail($"Invalid parameters for '{actionName}': {ex.Message}");
+            }
+            return handler(typed);
+        });
```
[the remark this answers is not in the document]
Comment (TalZaccai) [reply dotnet/autoShell/Handlers/ActionHandlerBase.cs:66]: Fixed in 5f1e70d5 — added explicit `ValueKind.Undefined/Null` guard before `GetRawText()`, returning `ActionResult.Fail` immediately. The null deserialization guard was already in place from a prior commit.

File: dotnet/autoShell/AutoShell.cs [hunk 3]
```diff
@@ -90,8 +91,7 @@ private static void RunFromCommandLine()
     /// </summary>
     private static void RunInteractive()
     {
-        bool quit = false;
-        while (!quit)
+        while (true)
         {
             try
             {
```
[the remark this answers is not in the document]
Comment (TalZaccai) [reply dotnet/autoShell/AutoShell.cs:106]: Fixed in 5f1e70d5 — catch block now writes a JSON `ActionResult.Fail` to stdout with the request id (extracted from the raw line if valid JSON) so the TS correlation resolves immediately.

File: dotnet/autoShell/AutoShell.cs [hunk 4]
```diff
@@ -104,8 +104,21 @@ private static void RunInteractive()
                 }
 
                 // Each line is a JSON object with one or more command keys
-                JObject root = JObject.Parse(line);
-                quit = ExecLine(root);
+                using var doc = JsonDocument.Parse(line);
+                var root = doc.RootElement;
+                ActionResult result = ExecLine(root);
```
[the remark this answers is not in the document]
Comment (TalZaccai) [reply dotnet/autoShell/AutoShell.cs:118]: Fixed in 5f1e70d5 — same fix (single catch block covers both parse and dispatch errors).

**Comment by robgruen** [comment]
Ship it!
